training/src/main/java/wolox/training/services/SecurityService.java

```diff
@@ -0,0 +1,42 @@
+package wolox.training.services;
+
+import com.google.common.collect.Sets;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.exceptions.UsernameExistsException;
+import wolox.training.models.User;
+import wolox.training.repositories.RoleRepository;
+import wolox.training.repositories.UserRepository;
+
+@Service
+public class SecurityService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    public User registerNewUserAccount(User user)
+        throws UsernameExistsException {
+
+        if (userRepository.findFirstByUsername(user.getUsername()).isPresent()) {
+            throw new UsernameExistsException
+                ("There is an account with that username: " + user.getUsername());
+        }
```

Comment:
Here you can use the benefits of the Optional too :) Like in the previous message